kukemc/I18n.js

```diff
@@ -0,0 +1,1239 @@
+        },
+      },
+    };
+
+    return {
+      id: kukemc_i18n_extensionId,
+      name: "I18n",
+      blockIconURI: kukemc_i18n_icon,
+      menuIconURI: kukemc_i18n_icon,
+      color1: "#13003a",
+      color2: "#0a001e",
+      docsURI:
+        "https://learn.ccw.site/article/99e0432c-98f2-4394-8a32-e501beee1e27",
+      blocks: [
+        "---" + this.formatMessage("kukeMCI18n.div.1"),
+        initI18nForJSON,
+        initI18nForURL,
+        initI18nForFile,
+        "---" + this.formatMessage("kukeMCI18n.div.2"),
+        getI18n,
+        getExtraData,
+        replaceText,
+        "---" + this.formatMessage("kukeMCI18n.div.3"),
+        setLanguage,
+        getLanguageForI18n,
+        getLanguageForBrowser,
+        getSupportedLanguages,
+        checkLanguageSupport,
+        "---" + this.formatMessage("kukeMCI18n.div.4"),
+        printSupportedLanguages,
+        "---" + this.formatMessage("kukeMCI18n.div.5"),
+        translateText,
+        translateTextWithClipboard,
+        "---" + this.formatMessage("kukeMCI18n.div.6"),
+        convertUnit,
+        getTimestamp,
+        getTimeZone,
+        offsetTimeZone,
+        formatTimestamp,
+      ],
+      menus: {
+        FILE_LIST: {
+          items: "__gandiAssetsFileList",
+        },
+        TRANSLATE_LIST: {
+          acceptReporters: true,
+          items: this.supportedTranslateCodes.map((t) => {
+            return {
+              text: `${t[0]}(${t[1]})`,
+              value: t[2],
+            };
+          }),
+        },
+        UNIT_LIST: {
+          acceptReporters: true,
+          items: this.supportedConvertibleUnits.map((t) => {
+            return {
+              text: `${t[0]}(${t[1]})`,
+              value: t[2],
+            };
+          }),
+        },
+      },
+    };
+  }
+
+  __gandiAssetsFileList() {
+    const list = this.runtime
+      .getGandiAssetsFileList("json")
+      .map((item) => item.name);
+    if (list.length < 1) {
+      list.push("没有文件 empty");
+    }
+
+    return list;
+  }
+
+  /**
+   * @param {String || Object} JSON
+   * @return {*}
+   */
+  initI18nForJSON({ JSON }) {
+    this.initialedI18n = JSON;
+  }
+
+  /**
+   * @param {String} URL
+   * @return {*}
+   */
+  initI18nForURL({ URL }) {
+    const xhr = new XMLHttpRequest();
+    xhr.open("GET", URL, false);
+    xhr.send();
+
+    if (xhr.status === 200) {
+      this.initialedI18n = xhr.responseText;
+    } else {
+      console.error(`[kukeMCI18n] request i18n failed: `);
+    }
+  }
+
+  /**
+   * @param {String} FILE
+   * @return {*}
+   */
+  initI18nForFile({ FILE }) {
+    const asset = this.runtime.getGandiAssetContent(FILE);
+    if (asset) {
+      this.initialedI18n = asset.decodeText();
+    }
+  }
+
+  /**
+   * @param {String} KEY
+   * @return {*}
+   */
+  getI18n({ KEY }) {
+    if (!this.i18n) return KEY;
+    return this.i18n[KEY];
+  }
+
+  getVariables() {
+    let variables = {};

```

Comment:
在 variables 本身不被重赋值时不需要将其设置为可变。
```suggestion
    const variables = {};
```